Bio/Seq.py

```diff
@@ -3080,6 +3124,22 @@ def lower(self):
         data = {start: seq.lower() for start, seq in self._data.items()}
         return _PartiallyDefinedSequenceData(self._length, data)
 
+    def isupper(self):
+        """Return True if all ASCII characters in data are uppercase.
+
+        If there are no cased characters, method returns False.
+        """
+        # Character case is irrelevant for an undefined sequence
+        raise UndefinedSequenceError("Sequence content is undefined")
```

Comment:
I would suggest `UndefinedSequenceError("Sequence content is only partially defined")` to be consistent with elsewhere in this class.